Task_One/IdealWeight.rb

```diff
@@ -0,0 +1,56 @@
+class CalculateIdealWeight
+  
+  class AttributeError < StandardError; end
+  SUBTRACTOR = 110
+  EDGE = 0
+
+  def initialize
+    print "Enter Your Name: "
+    name = gets.chomp
+    print "Enter Your Height: "
+    height = gets.chomp
+
+    @name = name
+    @height = height
+  end
+
+  def call
+    validate
+    calculate_ideal_weight
+    ideal_weight
+  rescue AttributeError => error
+    warn error
+  end
+
+  private
+
+  def ideal_weight
+    if @optimal_weight <= EDGE
+      print "#{@name} Your weight is already optimal"
+    else
+      print "#{@name} Your optimal weight is #{@optimal_weight}"
+    end
+  end
+
+  def calculate_ideal_weight
```

Comment:
rename it to ideal_weight

please read about Ruby Memoization
https://www.honeybadger.io/blog/ruby-rails-memoization/
https://www.justinweiss.com/articles/4-simple-memoization-patterns-in-ruby-and-one-gem/
https://thoughtbot.com/blog/ruby-memoization-and-alternatives

and use it in this method

call this method instead of using variable @optimal_weight